frontend/apps/filemanager/filemanager.lua

```diff
@@ -363,16 +363,30 @@ function FileManager:init()
 
     if Device:hasKeys() then
         self.key_events.Home = { {"Home"}, doc = "go home" }
-        if not Device:isSDL() then
-            --if not in the desktop emulator
-            --remove the old Back key to exit koreader
-            self.file_chooser.key_events.Close = nil
-        end
+        --Override the menu.lua way of handling the back key
+        self.file_chooser.key_events.Back = { {"Back"}, doc = "go back" }
     end
 
     self:handleEvent(Event:new("SetDimensions", self.dimen))
 end
 
+function FileChooser:onBack()
+    local back_to_exit = G_reader_settings:readSetting("back_to_exit") or "prompt"
+    if back_to_exit == "always" then
+        return self:onClose()
+    elseif back_to_exit == "disable" then
+        return true
+    elseif back_to_exit == "prompt" then
+            UIManager:show(ConfirmBox:new{
+                text ="Exit Koreader?",
+                ok_callback = function()
```

Comment:
Also
```lua
ok_text = _("Exit"),

Reply:
Sure, but I can't put _() here because it crash KOReader.

Reply:
With what error message?

Reply:
04/07/18-10:01:54 DEBUG painting widget: ReaderUI
04/07/18-10:01:55 DEBUG refresh on physical rectangle -1 -1 1002 1502
04/07/18-10:01:55 DEBUG key event => type: 1, code: 41(Back), value: 0, time: 1523088115.78814
04/07/18-10:01:55 DEBUG AutoSuspend: onInputEvent
04/07/18-10:01:55 DEBUG key event => type: 1, code: 41(Back), value: 1, time: 1523088115.809669
04/07/18-10:01:55 DEBUG AutoSuspend: onInputEvent
./luajit: frontend/apps/reader/modules/readerback.lua:87: attempt to call global '_' (a nil value)
stack traceback:
        frontend/apps/reader/modules/readerback.lua:87: in function 'handleEvent'
        frontend/ui/widget/container/widgetcontainer.lua:87: in function 'propagateEvent'
        frontend/ui/widget/container/widgetcontainer.lua:105: in function 'handleEvent'
        frontend/ui/uimanager.lua:454: in function 'sendEvent'
        frontend/ui/uimanager.lua:46: in function '__default__'
        frontend/ui/uimanager.lua:698: in function 'handleInputEvent'
        frontend/ui/uimanager.lua:759: in function 'handleInput'
        frontend/ui/uimanager.lua:803: in function 'run'
        ./reader.lua:195: in main chunk
        [C]: at 0x55a73e712050

Reply:
There's no `local _ = require("gettext")` in ReaderBack so it'd be very worrisome if you didn't get that error message.

What made it confusing is that you added this comment to the FileManager.

Reply:
Oh I see !
_ is actually a function !